### ISSUE TITLE
Rank Eval: Error on conflicting ratings on same document

### ISSUE THREAD

**Issue by cbuescher** [elastic/elasticsearch#20997]
I just ran into this when playing around with the discernatron ratings and generating new ranking requests. What behaviour do we expect when the user (mistakenly) provides several ratings for the same document, e.g. 

```
 "ratings" : [
{
          "_index" : "enwiki_rank",
          "_type" : "page",
          "_id" : "5119376",
          "rating" : 3
        },
       [....]
        {
          "_index" : "enwiki_rank",
          "_type" : "page",
          "_id" : "5119376",
          "rating" : 2
        }
```

This might happen unnoticed (like in my case) by some tool generating this part of the request from some external source. I think we should error on this, but one could also expect some sort of merging happening in this case (either taking the highest value, some average etc...). On the other hand, I wouldn't want merging of individual user/doc ratings happening within this api. @MaineC thoughts?

**Comment by MaineC** [comment]
I'd go for fail fast. It's unlikely that this was done on purpose so as a user I'd expect the system to tell me I've made a mistake.

**Comment by cbuescher** [comment]
Okay, I'm also for failing, we already do that currently but I think its later than parsing. We should probably already detect this when parsing the request.

**Comment by MaineC** [comment]
I think this can be closed now that #21146 is in.
